prov/gni/src/gnix_mr.c

```diff
@@ -440,6 +464,10 @@ int gnix_mr_reg(struct fid *fid, const void *buf, size_t len,
 		}
 	}
 
+	reg_addr = ((uint64_t) buf) & ~((1 << GNIX_MR_PAGE_SHIFT) - 1);
+	reg_len = __calculate_length((uint64_t) buf, len,
+			1 << GNIX_MR_PAGE_SHIFT);
```

Comment:
If the pagesize info in the gni memhandle is used, then we can't resize
the registration prior to calling GNI_MemRegister.

Additionally, there is no way to tell from the mem handle how long the
registration is. I've attempted to read the number of pages covered by the
memory handle and the number is bogus. It claims to have 353452 pages, all
4k size, or ~1.4GB for a 1k registration. That seems off to me.
-- Jim




On 11/3/15, 3:05 PM, "Howard Pritchard" <notifications@github.com> wrote:

>In 
>prov/gni/src/gnix_mr.c
><https://github.com/ofi-cray/libfabric-cray/pull/481#discussion_r43806617>
>:
>> @@ -440,6 +464,10 @@ int gnix_mr_reg(struct fid *fid, const void *buf,
>>size_t len,
>>  		}
>>  	}
>>  
>> +	reg_addr = ((uint64_t) buf) & ~((1 << GNIX_MR_PAGE_SHIFT) - 1);
>> +	reg_len = __calculate_length((uint64_t) buf, len,
>> +			1 << GNIX_MR_PAGE_SHIFT);
>i thought the rounding up/down was going to take in to account the
>pagesize info stored in the gni memhndl,
>—
>Reply to this email directly or
>view it on GitHub 
><https://github.com/ofi-cray/libfabric-cray/pull/481/files#r43806617>.
>
>